Mathlib/Data/List/Sort.lean

```diff
@@ -192,6 +191,108 @@ alias ⟨_, _root_.Monotone.ofFn_sorted⟩ := sorted_le_ofFn_iff
 
 end Monotone
 
+end List
+
+open List
+
+namespace StrictMono
+
+variable {α β : Type*} [LinearOrder α] [Preorder β] {f : α → β} {l : List α}
+
+theorem sorted_le_listMap (hf : StrictMono f) :
+    (l.map f).Sorted (· ≤ ·) ↔ l.Sorted (· ≤ ·) := by
+  simp only [Sorted, pairwise_map, hf.le_iff_le]
+
+theorem sorted_ge_listMap (hf : StrictMono f) :
+    (l.map f).Sorted (· ≥ ·) ↔ l.Sorted (· ≥ ·) := by
+  simp only [Sorted, pairwise_map, hf.le_iff_le]
+
+theorem sorted_lt_listMap (hf : StrictMono f) :
+    (l.map f).Sorted (· < ·) ↔ l.Sorted (· < ·) := by
+  simp only [Sorted, pairwise_map, hf.lt_iff_lt]
+
+theorem sorted_gt_listMap (hf : StrictMono f) :
+    (l.map f).Sorted (· > ·) ↔ l.Sorted (· > ·) := by
+  simp only [Sorted, pairwise_map, hf.lt_iff_lt]
+
+end StrictMono
+
+namespace StrictAnti
+
+variable {α β : Type*} [LinearOrder α] [Preorder β] {f : α → β} {l : List α}
+
+theorem sorted_le_listMap (hf : StrictAnti f) :
+    (l.map f).Sorted (· ≤ ·) ↔ l.Sorted (· ≥ ·) := by
+  simp only [Sorted, pairwise_map, hf.le_iff_le]
+
+theorem sorted_ge_listMap (hf : StrictAnti f) :
+    (l.map f).Sorted (· ≥ ·) ↔ l.Sorted (· ≤ ·) := by
+  simp only [Sorted, pairwise_map, hf.le_iff_le]
+
+theorem sorted_lt_listMap (hf : StrictAnti f) :
+    (l.map f).Sorted (· < ·) ↔ l.Sorted (· > ·) := by
+  simp only [Sorted, pairwise_map, hf.lt_iff_lt]
+
+theorem sorted_gt_listMap (hf : StrictAnti f) :
+    (l.map f).Sorted (· > ·) ↔ l.Sorted (· < ·) := by
+  simp only [Sorted, pairwise_map, hf.lt_iff_lt]
+
+end StrictAnti
+
+namespace OrderEmbedding
+
+variable {α β : Type*} [Preorder α] [Preorder β]
+
+@[simp]
+theorem sorted_le_listMap (e : α ↪o β) {l : List α} :
```

Comment:
Is there a version of this for arbitrary relation embeddings, or are we missing the general version of `Preorder`?

Reply:
Indeed, this works fine:
```lean
namespace RelEmbedding

variable {α β : Type*} (r : α → α → Prop) (s : β → β → Prop)

@[simp]
theorem sorted_listMap (e : r ↪r s) {l : List α} :
    (l.map e).Sorted s ↔ l.Sorted r := by
  simp only [Sorted, pairwise_map, e.map_rel_iff]
```